overlay/agent.cpp

```diff
@@ -207,12 +167,70 @@ Try<Owned<ManagerProcess>> ManagerProcess::create(
     }
   }
 
+  Try<Owned<MasterDetector>> detector =
+    ManagerProcess::createDetector(agentConfig, _detector);
+  if (detector.isError()) {
+    return Error(detector.error());
+  }
+
   return Owned<ManagerProcess>(
       new ManagerProcess(
         agentConfig.cni_dir(),
         networkConfig,
         agentConfig.max_configuration_attempts(),
-        Owned<MasterDetector>(detector.get())));
+        detector.get()));
+}
+
+
+Try<Owned<MasterDetector>> ManagerProcess::createDetector(
+    const AgentConfig& agentConfig,
+    Option<Owned<MasterDetector>> _detector)
```

Comment:
Lets remove the `_detector` argument, it seems redundant, since we return the `Owned` contained the `Option` if the `Option` is set. We can just check for the `_detector` when we invoke `createDetector`. Please see my comments above.